### ISSUE TITLE
Demolition of buildings: Abbot's cathedral

### ISSUE THREAD

**Issue by Heroesflorian** [Sh0wdown/UnofficialCrusaderPatch#219]
There is an option in the patch that prevents the AI from demolishing certain buildings when trying to gather gold when in danger / deep financial crisis. 

However, this does not include the Abbot's cathedral which will still be demolished, rendering him nearly defenseless without any way to recruit melee troops...

**Comment by PitchNeeded** [comment]
The Abbot is very rarely in deep financial crisis anyway (being the best economist in the game), but if he ever is, yes his cathedral shouldn't be demolished on no AI demolishing buildings mode.

**Comment by Heroesflorian** [comment]
@PitchNeeded Financial crisis or not, the AI also triggers the "demolish buildings" routine when in danger / fear, which happens when an attacking army substantially outnumbers the currently available own units of the defender. 
I've seen the Abbot demolish buildings due to that even when having over 50k gold reserves, when he had just attacked (and thus had no own troops waiting on his keep anymore), and then got counter-attacked right after that. 
This behaviour then had the consequence of the Abbot not being able to recruit monks... uargh... :D 
The Abbot not recruiting monks (but only archers) when in fear is another issue, but destroying his cathedral without any reason is just soo pointless. 

(And yes, fear-induced building demolition in general is not specific to the Abbot - it can happen for any AI character. Abbot just happens to be the only one using cathedrals for monks.)
